torchvision/csrc/nms.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include "cpu/vision.h"
+
+#ifdef WITH_CUDA
+#include "cuda/vision.h"
+#endif
+
+at::Tensor nms(
+    const at::Tensor& dets,
+    const at::Tensor& scores,
+    const float threshold) {
+  if (dets.device().is_cuda()) {
+#ifdef WITH_CUDA
+    if (dets.numel() == 0) {
+      at::cuda::CUDAGuard device_guard(dets.device());
+      return at::empty({0}, dets.options().dtype(at::kLong));
+    }
+    auto b = at::cat({dets, scores.unsqueeze(1)}, 1);
```

Comment:
I've been wondering about this. It seems unnecessary to waste memory creating a new tensor here when we can easily just update the function signature to accept 3 arguments. Thoughts?

Reply:
I totally agree, I just wanted to minimize the amount of work for now, but this is definitely something I'd like to have done

Reply:
I'll add a PR on top of this branch to add that later today.